staging/src/k8s.io/client-go/examples/third-party-resources/main.go

```diff
@@ -78,19 +77,29 @@ func main() {
 	}
 
 	// make a new config for our extension's API group, using the first config as a baseline
-	var tprconfig *rest.Config
-	tprconfig = config
-	configureClient(tprconfig)
-
-	tprclient, err := rest.RESTClientFor(tprconfig)
+	exampleClient, exampleScheme, err := NewClient(config)
 	if err != nil {
 		panic(err)
 	}
 
+	// start a watcher on instances of our TPR
+	watcher := Watcher{
+		clientset:     clientset,
+		exampleClient: exampleClient,
+		exampleScheme: exampleScheme,
+	}
+
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	defer cancelFunc()
+	go watcher.Run(ctx)
+
+	time.Sleep(5 * time.Second) // Wait until watcher starts
```

Comment:
@sttts I've added the following comment:
```
	// The sleep below is just to make sure that the watcher.Run() goroutine has successfully executed
	// and the watcher is handling the events about Example TPR instances.
	// In the normal application there is no need for it, because:
	// 1. It's unlikely to create a watcher and a TPR instance at the same time in the same application.
	// 2. The application with watcher would most probably keep running instead of exiting right after the watcher startup.
```